submitr/submission.py

```diff
@@ -1891,11 +1910,14 @@ def progress_report(status: dict) -> None:  # noqa
                                         # i.e. this is really remove empty trailing object from arrays.
                                         remove_empty_objects_from_lists=True,
                                         # If the --merge option is given then merge the
-                                        # given (i.e. e.g. spreasheet) object(s) into any existing ones.
+                                        # given (i.e. e.g. spreadsheet) object(s) into any existing ones.
                                         merge=merge,
                                         progress=None if noprogress else define_progress_callback(debug=debug),
+                                        # Doing submitted_id validation in validate_submitted_ids, in parallel.
+                                        validator_hook=validator_hook,
```

Comment:
This is the main generic hook into the submission metadata parsing (which currently used just to validate submission_id across all schema/types/sheets).